libraries/provider_rack_user.rb

```diff
@@ -1,4 +1,5 @@
 require 'chef/resource/lwrp_base'
+require 'shadow'
```

Comment:
I think you both have a valid point, but as raised by @seeafish , it's hard to know every dependencies(if they are already met by Chef core) and make sure they are installed before Chef converge.

I assume always forcing shadow gem will make the converge a bit slower too :-P